### PR TITLE
feat: esp32-camera add stream event

### PR DESCRIPTION
**Pull request by myml**
# What does this implement/fix?

esp32-camera add 'on_stream_start' and 'on_stream_stop' stream event for use in Automotions.
For example, the built-in flash led turn on automatically while viewing.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Related issue or feature (if applicable):** fixes <link to issue>

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#2078

## Test Environment

- [x] ESP32
- [ ] ESP32 IDF
- [ ] ESP8266

## Example entry for `config.yaml`:

```yaml
# Example config.yaml
esp32_camera:
  external_clock:
    pin: GPIO0
    frequency: 20MHz
  i2c_pins:
    sda: GPIO26
    scl: GPIO27
  data_pins: [GPIO5, GPIO18, GPIO19, GPIO21, GPIO36, GPIO39, GPIO34, GPIO35]
  vsync_pin: GPIO25
  href_pin: GPIO23
  pixel_clock_pin: GPIO22
  power_down_pin: GPIO32
  name: camera1
  id: camera1
  vertical_flip: false
  on_stream_start:
    - script.execute: open_flash_script

script:
  - id: open_flash_script
    mode: restart
    then:
      - if:
          condition:
            light.is_off: cam_light
          then:
            - light.turn_on: cam_light
      - delay: 3s
      - light.turn_off: cam_light

light:
  - id: cam_light
    platform: binary
    name: "cam1 light"
    output: light_output
    
output:
  - id: light_output
    platform: gpio
    pin: GPIO4
```

## Checklist:
  - [x] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

File: esphome/components/esp32_camera/__init__.py [hunk 1]
```diff
@@ -111,6 +113,19 @@
 CONF_MAX_FRAMERATE = "max_framerate"
 CONF_IDLE_FRAMERATE = "idle_framerate"
 
+# stream trigger
+CONF_ON_STREAM_START = "on_stream_start"
+CONF_ON_STREAM_STOP = "on_stream_stop"
+ec_ns = cg.esphome_ns.namespace("esp32_camera")
```
Comment (jesserockz) [inline esphome/components/esp32_camera/__init__.py]: This already exists above as `esp32_camera_ns`
```suggestion
```

**Review by jesserockz — APPROVED** [review]
One minor thing, otherwise looks good

File: esphome/components/esp32_camera/__init__.py (hunk 1 above)
Comment (myml) [reply esphome/components/esp32_camera/__init__.py]: Thank you, fixed.

**Review by jesserockz — APPROVED** [review]

**Comment by jesserockz** [comment]
@myml Do you have a docs PR for this?

**Comment by myml** [comment]
> @myml Do you have a docs PR for this?

No, can you help me add?

**Comment by jesserockz** [comment]
https://esphome.io/guides/contributing.html#contributing-to-esphome-docs

**Comment by myml** [comment]
@jesserockz  docs has been added https://github.com/esphome/esphome-docs/pull/2078

**Pull request merged**